### ISSUE TITLE
Referencing an empty patch file throws panic

### ISSUE THREAD

**Issue by fentas** [kubernetes-sigs/kustomize#1060]
```
patchesJson6902:
- target: { group: apps, version: v1beta1, kind: Deployment, name: rook-ceph-operator }
  path: patches/rook-ceph-operator.yaml
```

if `patches/rook-ceph-operator.yaml` is empty it will throw
```sh
➜ kustomize build kustomize/infrastructure/overlays/hetzner/
panic: runtime error: index out of range

goroutine 1 [running]:
sigs.k8s.io/kustomize/pkg/patch/transformer.isJsonFormat(...)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/patch/transformer/factory.go:82
sigs.k8s.io/kustomize/pkg/patch/transformer.newPatchJson6902JSONTransformer(0xc0004a1378, 0x4, 0xc0004a1390, 0x2, 0xc0004a1380, 0x9, 0xc0004a138a, 0x6, 0x0, 0x0, ...)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/patch/transformer/patchjson6902json.go:45 +0x296
sigs.k8s.io/kustomize/pkg/patch/transformer.PatchJson6902Factory.makeOnePatchJson6902Transformer(0x1355460, 0xc0004ba0c0, 0xc0000e04b0, 0xc0004ab720, 0x13, 0xc0003b3690, 0x0, 0x1, 0x0)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/patch/transformer/factory.go:78 +0x2f7
sigs.k8s.io/kustomize/pkg/patch/transformer.PatchJson6902Factory.MakePatchJson6902Transformer(0x1355460, 0xc0004ba0c0, 0xc0000d2cc0, 0x2, 0x4, 0xc00045dc80, 0x4, 0x8, 0x0)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/patch/transformer/factory.go:43 +0xc7
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).newTransformer(0xc00049e510, 0x0, 0x0, 0x0, 0xc00068e790, 0x0, 0x0, 0x0, 0x0)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/target/kusttarget.go:304 +0x5f4
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).AccumulateTarget(0xc00049e510, 0xc0000f9c48, 0xc000052580, 0xc0000e0080)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/target/kusttarget.go:205 +0x46f
sigs.k8s.io/kustomize/pkg/target.(*KustTarget).MakeCustomizedResMap(0xc00049e510, 0xc0004ba0c0, 0xc0000d1738, 0x133ab80)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/target/kusttarget.go:122 +0x2f
sigs.k8s.io/kustomize/pkg/commands/build.(*Options).RunBuild(0xc0003cfc60, 0x13303e0, 0xc0000d0008, 0x1375c80, 0x1cbb0c0, 0xc0000d1738, 0x133ab80, 0x1cbb0c0, 0x0, 0x0)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/commands/build/build.go:115 +0x108
sigs.k8s.io/kustomize/pkg/commands/build.NewCmdBuild.func1(0xc0001e2c80, 0xc00002e0a0, 0x1, 0x1,
0x0, 0x0)
        /home/user/go/pkg/mod/sigs.k8s.io/kustomize@v2.0.3+incompatible/pkg/commands/build/build.go:78 +0x136
github.com/spf13/cobra.(*Command).execute(0xc0001e2c80, 0xc00002e060, 0x1, 0x1, 0xc0001e2c80, 0xc00002e060)
        /home/user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc0001e2a00, 0xc0001e2a00, 0xf8d120, 0x11eedfc)
        /home/user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2c0
github.com/spf13/cobra.(*Command).Execute(...)
        /home/user/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800
main.main()
        /home/user/go/pkg/mod/github.com/kubernetes-sigs/kustomize@v2.0.3+incompatible/kustomize.go:27 +0x15c
```

**Comment by Liujingfang1** [comment]
We should fix this issue before the 2.1 release.

**Comment by Liujingfang1** [comment]
Should be fixed by #1073
